File: server/src/main/java/org/dan/jadalnia/app/festival/menu/DishName.java
package org.dan.jadalnia.app.festival.menu;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.ToString;


@ToString
@EqualsAndHashCode
@Getter(onMethod = @__(@JsonValue))
@RequiredArgsConstructor(onConstructor = @__(@JsonCreator))
public class DishName {
    private final String name;
}

File: kot/src/main/kotlin/org/dan/jadalnia/app/festival/menu/DishName.kt
package org.dan.jadalnia.app.festival.menu;

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

data class DishName constructor(val name: String): Comparable<DishName> {
    companion object {
        @JvmStatic
        @JsonCreator
        fun of(s: String) = DishName(s)
    }

    @JsonValue
    override fun toString(): String = name

    override fun compareTo(other: DishName) = name.compareTo(other.name)
}

